File: Introductory Problems/8. two sets/two_sets.py
def two_sets(n):
    set1 = []
    set2 = []
    full_set = [i+1 for i in range(n)]
    list_sum = sum(full_set)
    full_set.reverse()
    if list_sum % 2 != 0:
        print('NO')
    else:
        mid_value = list_sum / 2
        for num in full_set:
            if num <= mid_value:
                set1.append(num)
                mid_value -= num
            else:
                set2.append(num)
        print(len(set1), set1)
        print(len(set2), set2)


two_sets(8)

File: Introductory Problems/two sets/twoSets.js
function twoSets(n) {
  const set1 = [],
    set2 = [];
  const fullArray = Array.from({ length: n }, (_, i) => i + 1).reverse();
  const arraySum = fullArray.reduce((acc, curr) => curr + acc, 0);
  if (arraySum % 2 != 0) return "NO";
  else {
    console.log("YES");
    let midValue = arraySum / 2;
    for (let i = 0; i < n; i++) {
      if (fullArray[i] <= midValue) {
        set1.push(fullArray[i]);
        midValue -= fullArray[i];
      } else {
        set2.push(fullArray[i]);
      }
    }
  }
  console.log(set1.length, set1);
  console.log(set2.length, set2);
}

twoSets(10);
